example/plugins/c-api/denylist_1/denylist_1.c

```diff
@@ -179,7 +179,7 @@ read_denylist(TSCont contp)
   char denylist_file[1024];
   TSFile file;
 
-  sprintf(denylist_file, "%s/denylist.txt", TSPluginDirGet());
+  snprintf(denylist_file, 1024, "%s/denylist.txt", TSPluginDirGet());
```

Comment:
nitpick:  in this case you could use `sizeof(denylist_file)` so the size only needs to be changed in on place.